crates/starknet-devnet/tests/test_blocks_generation.rs

```diff
@@ -412,6 +418,64 @@ mod blocks_on_demand_tests {
         assert_eq!(get_contract_balance(&devnet, contract_address).await, expected_balance);
     }
 
+    #[tokio::test]
+    async fn blocks_on_interval() {
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&["--block-generation-on", "1"])
+            .await
+            .expect("Could not start Devnet");
+
+        // wait 1 second
+        tokio::time::sleep(time::Duration::from_secs(1)).await;
+
+        let last_block = devnet.get_latest_block_with_tx_hashes().await.unwrap();
+
+        // first is genesis block, second block is generated instantly, third is generated after 1
+        // second
+        assert_eq!(last_block.block_number, 2);
+    }
+
+    #[tokio::test]
+    async fn blocks_on_interval_dump_and_load() {
+        let mode = "exit";
+        let dump_file = UniqueAutoDeletableFile::new(
+            ("blocks_on_interval_dump_and_load_".to_owned() + mode).as_str(),
+        );
+
+        let devnet_dump = BackgroundDevnet::spawn_with_additional_args(&[
+            "--dump-path",
+            &dump_file.path,
+            "--dump-on",
+            mode,
+            "--block-generation-on",
+            "1",
```

Comment:
TBH this one second interval seems like it could lead to flaky tests in CircleCI (since we have exact block number assertion). On the other hand, I see you've executed the workflow multiple times with no such errors.

Reply:
I tested that locally and on ci/cd, if that problem occurs we can add >= assertion, but for now I would leave it as it is